Setting up repository leads to dependency error

Hi, when I follow the directions to clone the repo:
```
# clone repo with submodules
git clone --recursive git@github.com:ocaml/ocaml-lsp.git

cd ocaml-lsp

# if you already cloned, pull submodules
git submodule update --init --recursive

# create local switch (or use global one)
opam switch --yes create . ocaml-base-compiler.4.14.0

# don't forget to set your environment to use the local switch
eval $(opam env)

# install dependencies
make install-test-deps

# build
make all

# the ocamllsp executable can be found at _build/default/ocaml-lsp-server/bin/main.exe
```

at the step where I create a local switch, this happens:
```
<><> Installing new switch packages <><><><><><><><><><><><><><><><><><><><><><>
Switch invariant: ["ocaml-base-compiler" {= "4.14.0"}]
[ERROR] Could not determine which packages to install for this switch:
  * No agreement on the version of ocaml-base-compiler:
    - (invariant) → ocaml-base-compiler = 4.14.0
    - ocaml-lsp-server → ocaml >= 5.2.0 → ocaml-base-compiler >= 5.2.0~
    You can temporarily relax the switch invariant with `--update-invariant'
  * Missing dependency:
    - ocaml-lsp-server → ocaml >= 5.2.0 → ocaml-variants >= 5.2.0~ → system-msvc
    unmet availability conditions: 'os = "win32"'
```
It seems like the project uses `ocaml-lsp-server` as a dependency, but it doesn't support compiler version 4.14.0.

I am not too familiar with the OCaml ecosystem, but would there be any reason to not pass the `--update-invariant` flag? What does it do exactly?

--- a/README.md
+++ b/README.md
@@ -319,7 +319,7 @@ type point = {x: int; y: int}
 Hovering over the above will
 display:
 ```
-ocaml type point = { x : int; y : int } 
+ocaml type point = { x : int; y : int }
 syntax Record type:
 Allows you to define variants with a fixed set of fields, and all of the
 constructors for a record variant type must have the same fields. See
@@ -328,7 +328,7 @@ Manual
 The documentation is gotten from the Merlin engine which receives
 the nodes under the cursor and infers what the syntax may be about, and
 displays the required information along with links to the manual for further
-reading. 
+reading.
 
 Syntax Documentation is an optional feature and can be activated by
 using the LSP config system with the key called `syntaxDocumentation` and can
@@ -356,7 +356,7 @@ cd ocaml-lsp
 git submodule update --init --recursive
 
 # create local switch (or use global one)
-opam switch --yes create . ocaml-base-compiler.4.14.0
+opam switch --yes create .
 
 # don't forget to set your environment to use the local switch
 eval $(opam env)

